napari/_qt/qt_resources/_icons.py

```diff
@@ -0,0 +1,317 @@
+            if 'pyside2' not in bin:
+                # the newer pure rcc version requires this for python
+                cmd.extend(['-g', 'python'])
+            break
+    else:
+        raise RuntimeError(f"PySide2 rcc binary not found in {pyside_root}")
+
+    try:
+        return run(cmd + [qrc], check=True, capture_output=True).stdout
+    except CalledProcessError as e:
+        raise RuntimeError(f"Failed to build PySide2 resources {e}")
+
+
+def compile_qrc(qrc) -> bytes:
+    """Compile a qrc file into a resources.py bytes"""
+    if qtpy.API_NAME == 'PyQt5':
+        return _compile_qrc_pyqt5(qrc).replace(b'PyQt5', b'qtpy')
+    elif qtpy.API_NAME == 'PySide2':
+        return _compile_qrc_pyside2(qrc).replace(b'PySide2', b'qtpy')
+    else:
+        raise RuntimeError(
+            f"Cannot compile QRC. Unexpected qtpy API name: {qtpy.API_NAME}"
+        )
+
+
+def compile_qt_svgs(
+    svg_paths: Iterable[Union[str, Path]],
+    colors: Iterable[Union[str, Tuple[str, str]]],
+    opacities: Iterable[float] = (1.0,),
+    theme_override: Optional[Dict[str, str]] = None,
+    prefix='',
+    save_path: Optional[str] = None,
+    register: bool = False,
+) -> str:
+    """[summary]
+
+    Parameters
+    ----------
+    svg_paths : Iterable[Union[str, Path]]
+        An iterable of paths to svg files
+    colors : Iterable[Union[str, Tuple[str, str]]]
+        An iterable of colors.  Every icon will be generated in every color. If
+        a `color` item is a string, it should be valid svg color style. Items
+        may also be a 2-tuple of strings, in which case the first item should
+        be an available theme name
+        (:func:`~napari.utils.theme.available_themes`), and the second item
+        should be a key in the theme (:func:`~napari.utils.theme.get_theme`),
+    opacities : Iterable[float], optional
+        An iterable of opacities to generate, by default (1.0,) Opacities less
+        than one can be accessed in qss with the opacity as a percentage
+        suffix, e.g.: `my_svg_50.svg` for opacity 0.5.
+    theme_override : dict, optional
+        When one of the `colors` is a theme ``(name, key)`` tuple,
+        `theme_override` may be used to override the `key` for a specific icon
+        name in `svg_paths`.  For example `{'exclamation': 'warning'}`, would
+        use the theme "warning" color for any icon named "exclamation.svg" by
+        default None
+    prefix : str, optional
+        A prefix to use when accessing these resources.  For instance, if
+        ``prefix='theme'``, and one of the aliases in `xmls` is
+        `"my_icon.svg"`, then you could access the compiled resources at
+        `:/theme/my_icon.svg`. by default ''
+    save_path : str, optional
+        An optional output path for the compiled py resources, by default None
+    register : bool, optional
+        If `True`, immediately execute (import) the compiled resources.
+        By default `False`
```

Comment:
Why would someone use register=False / why is default False? It looks like both our internal usage and usage in example has register=True.

Reply:
registration is a bit "heavy".  It demands that `QApplication` is running (and starts one if it isn't) and actively executes bytes (importing the resource file).  So, I thought it safest here to request that the user explicitly request it.  

> Why would someone use register=False

To compile and potentially save a `_qt_resources.py` file for later usage.  That was basically the default of our previous compile process to.